### DISCUSSION TITLE
[Bug] Multiple files/urls of a program are handled in different ways

### DISCUSSION THREAD

**Discussion by Nashmatphysics** [ScoopInstaller/Scoop#6222] [Help and Questions]
env:
os: windows 10 21h2
pwsh: 7.4

hi.
When I write a manifest for a program, there is a problem I can't solve, like this:
```json
{
    "url": [
        "https://github.com/app.exe",
        "https://github.com/source.zip"
    ],
    "hash": [
        "bf3a27398397b886a3d8b0ad2591c29a7165d1f560b121c4c03ed331657d3c4",
        "287079ba96dcfa79fa6c568481f4a26bd3ac26671f8a21c4c03ed331657d53c0"
    ], 
    "innosetup": true,
}
```

I will get an error, because scoop will send both the exe and zip files to innounp for processing, and innounp can't deal with zip file.
So how do I specify that the exe file uses innounp and the zip file does not use innounp?
Is there a way to configure each url separately to decompress it?

Thank you.

**Comment by HUMORCE** [comment]
Remove `innosetup` prop, then extract inno installer with `Expand-InnoArchive` on `pre_install`.

https://github.com/ScoopInstaller/Scoop/blob/859d1db51bcc840903d5280567846ae2f7207ca2/lib/decompress.ps1#L226
